python/cudf/cudf/core/dataframe.py

```diff
@@ -3785,6 +3785,60 @@ def to_csv(
             chunksize,
         )
 
+    def scatter_by_map(self, map_index, map_size=None):
+        """Scatter to a list of dataframes.
+
+        Uses map_index to determine the destination
+        of each row of the original DataFrame.
+
+        Parameters
+        ----------
+        map_index : Series, str or list-like
+            Scatter assignment for each row
+        map_size : int
+            Length of output list. Must be >= uniques in map_index
+
+        Returns
+        -------
+        A list of cudf.DataFrame objects.
+        """
+
+        # map_index might be a column name or array,
+        # make it a Series
+        if isinstance(map_index, str):
+            map_index = self[map_index]
+        else:
+            map_index = Series(map_index)
+
+        # Convert float to integer
+        if map_index.dtype == np.float:
+            map_index = map_index.astype(np.int32)
+
+        # Convert string or categorical to integer
+        if isinstance(map_index._column, StringColumn):
+            map_index = Series(
+                map_index._column.as_categorical_column(np.int32).as_numerical
+            )
+        elif isinstance(map_index._column, CategoricalColumn):
+            map_index = Series(map_index._column.as_numerical)
+
+        # scatter_to_frames wants a list of Series/columns
+        source = [self[col] for col in self.columns]
```

Comment:
We have a helper function for this already:
```suggestion
        source = self._columns
```

Reply:
Interesting - This change actually breaks the code because `_columns()` resets the name of each column in the list to `None`.

Reply:
That may be a remnant from when names weren't properly handled in libcudf. You can either track the names separately and adjust or try to change that function, but unsure if something else will break.

Reply:
Okay - Looks like changing `_columns` doesn't break any tests :)